File: src/main/java/com/jss/osiris/modules/quotation/service/NoticeTypeFamilyService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.NoticeTypeFamily;

public interface NoticeTypeFamilyService {
    public List<NoticeTypeFamily> getNoticeTypeFamilies();

    public NoticeTypeFamily getNoticeTypeFamily(Integer id);
}

File: client-myjss/src/app/modules/quotation/services/notice.type.family.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from '../../main/services/appRest.service';
import { NoticeTypeFamily } from '../model/NoticeTypeFamily';

@Injectable({
  providedIn: 'root'
})
export class NoticeTypeFamilyService extends AppRestService<NoticeTypeFamily> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getNoticeTypeFamilies() {
    return this.getListCached(new HttpParams(), "notice-type-families");
  }

}
